source/credentials_provider_profile.c

```diff
@@ -45,52 +46,60 @@ static int s_profile_file_credentials_provider_get_credentials_async(
 
     struct aws_credentials_provider_profile_file_impl *impl = provider->impl;
     struct aws_credentials *credentials = NULL;
+    struct aws_profile_collection *merged_profiles = aws_profile_collection_acquire(impl->profile_collection_cached);
 
-    /*
-     * Parse config file, if it exists
-     */
-    struct aws_profile_collection *config_profiles =
-        aws_profile_collection_new_from_file(provider->allocator, impl->config_file_path, AWS_PST_CONFIG);
+    if (!merged_profiles) {
+        /*
+         * Parse config file, if it exists
+         */
+        struct aws_profile_collection *config_profiles =
+            aws_profile_collection_new_from_file(provider->allocator, impl->config_file_path, AWS_PST_CONFIG);
 
-    if (config_profiles != NULL) {
-        AWS_LOGF_DEBUG(
-            AWS_LS_AUTH_CREDENTIALS_PROVIDER,
-            "(id=%p) Profile credentials provider successfully built config profile collection from file at (%s)",
-            (void *)provider,
-            aws_string_c_str(impl->config_file_path));
-    } else {
-        AWS_LOGF_DEBUG(
-            AWS_LS_AUTH_CREDENTIALS_PROVIDER,
-            "(id=%p) Profile credentials provider failed to build config profile collection from file at (%s)",
-            (void *)provider,
-            aws_string_c_str(impl->config_file_path));
-    }
+        if (config_profiles != NULL) {
+            AWS_LOGF_DEBUG(
+                AWS_LS_AUTH_CREDENTIALS_PROVIDER,
+                "(id=%p) Profile credentials provider successfully built config profile collection from file at (%s)",
+                (void *)provider,
+                aws_string_c_str(impl->config_file_path));
+        } else {
+            AWS_LOGF_DEBUG(
+                AWS_LS_AUTH_CREDENTIALS_PROVIDER,
+                "(id=%p) Profile credentials provider failed to build config profile collection from file at (%s)",
+                (void *)provider,
+                aws_string_c_str(impl->config_file_path));
+        }
```

Comment:
Should we log that we are using merged_profiles (profile_collection_cached) and its values here?